asynction/server.py

```diff
@@ -155,7 +157,12 @@ def from_spec(
             )
 
         """
-        spec = load_spec(spec_path=spec_path)
+        if isinstance(spec_path, Path):
+            spec = load_spec(spec_path=spec_path)
+        else:
+            raw_resolved = resolve_references(raw_spec=spec_path)
+            spec = AsyncApiSpec.from_dict(raw_resolved)
```

Comment:
I think it would be neater if we could move this logic within the existing `load_spec` function:

```python
def load_spec(raw_spec: Union[Path, JSONMapping]) -> AsyncApiSpec:
    if isinstance(spec, Path): 
        with open(spec_path) as f:
            serialized = f.read()
            raw_spec = yaml.safe_load(serialized)

    raw_resolved = resolve_references(raw_spec)
    return AsyncApiSpec.from_dict(raw_resolved)
```

Note that here, in the `load_spec` method, we can change the argument name from `spec_path` to `raw_spec` since `load_spec` is not part of Asynction's "public" API.

Reply:
The above line in the `from_spec` method would then become:
```suggestion
     spec = load_spec(raw_spec=spec_path)
```